src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -111,6 +115,9 @@ protected DefaultConfiguration(IEnvironment environment, configuration localConf
 
             ParseExpectedErrorConfigurations();
             ParseIgnoreErrorConfigurations();
+
+            _hasObscuringKey = !string.IsNullOrEmpty(ObscuringKey);
+            _hasEncryptedPassword = !string.IsNullOrEmpty(_localConfiguration.service.proxy.encryptedPassword);
```

Comment:
I don't think we need this.  Probably can move this logic into the if statements